pkg/logger/ecs_error_log_test.go

```diff
@@ -0,0 +1,94 @@
+package logger
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+)
+
+var testErrorLogFilePath = "/opt/aws/aws-otel-collector/logs/aws-otel-collector-error-test.log"
+
+func TestWriteError(t *testing.T) {
+	defer os.Remove(testErrorLogFilePath)
+	openExistingOrNewErrorFile = OpenExistOrNew
+	entry := zapcore.Entry{
+		Message: "test",
+	}
+	_, error := WriteError(entry, testErrorLogFilePath)
+	require.NoError(t, error)
+	queue.Remove(queue.Back())
+	b, err := ioutil.ReadFile(testErrorLogFilePath)
+	if err != nil {
+		return
+	}
+	s := string(b)
+	assert.True(t, strings.Contains(s, "test"))
+}
+
+func TestGetErrorLogFilePath(t *testing.T) {
+	logPath := getErrorLogFilePath()
+	if runtime.GOOS == "windows" {
+		assert.Equal(t, WindowsErrorLogPath, logPath)
+	} else {
+		assert.Equal(t, UnixErrorLogPath, logPath)
+	}
+}
+
+func TestRotate(t *testing.T) {
+	defer os.Remove(testErrorLogFilePath)
+	testErrorLogFile, _ := os.OpenFile(testErrorLogFilePath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC|os.O_APPEND, 0777)
+	errormessage1 := make([]byte, 700)
+	errormessage2 := make([]byte, 700)
+	for i := 0; i < 700; i++ {
+		errormessage1[i] = 'a'
+		errormessage2[i] = 'b'
+	}
+	e1 := zapcore.Entry{
+		Message: string(errormessage1),
+	}
+	e2 := zapcore.Entry{
+		Message: string(errormessage2),
+	}
+	content1 := []byte(fmt.Sprintf("{%+v, Level:%+v, Caller:%+v, Message:%+v}\r\n",
+		e1.Time, e1.Level, e1.Caller, e1.Message))
+	content2 := []byte(fmt.Sprintf("{%+v, Level:%+v, Caller:%+v, Message:%+v}\r\n",
+		e2.Time, e2.Level, e2.Caller, e2.Message))
+	_, _ = testErrorLogFile.Write(content1)
+	_, _ = testErrorLogFile.Write(content2)
+	queue.PushBack(e1)
+	queue.PushBack(e2)
+	byteNum := len(content1) + len(content2)
+	n, err := rotate(byteNum, testErrorLogFilePath)
+	require.NoError(t, err)
+	assert.Equal(t, len(content2), n)
+	queue.Remove(queue.Back())
+}
+
+func OpenExistOrNew() error {
```

Comment:
This function seems unnecessary to me. For unit test, it's better for you to use the original function which you used in your code instead of creating a new function here.

Reply:
I agree with John's comment. 

Can't you use the env vars in the test? You can set the test file path as the value of the env var and then the code will use it. 

This will work if you do my suggested refactoring where you move all global variables into a structure.